packages/sitecore-jss-nextjs/src/utils.ts

```diff
@@ -75,3 +75,22 @@ export const getJssEditingSecret = (): string => {
   }
   return secret;
 };
+
+/**
+ * Parses raw point of sale value in Record (locale-bound or default)
+ * if we have a valid multi-value JSON input we return a Record
+ * if POS input is single value - return Record with a single value in it
+ * point of sale is just a string from CDP's point of view, not URL neccessarily - so no validation
+ * @returns non-empty Record
+ */
+export const parsePointOfSaleRawInput = (cdpPos: string): Record<string, string> => {
+  if (cdpPos.startsWith('{')) {
+    try {
+      return JSON.parse(cdpPos) as Record<string, string>;
+    } catch (error) {
+      return { default: cdpPos } as Record<string, string>;
```

Comment:
We should probably log the error so they're aware that formatting is incorrect.
Also, I'm not sure we want to return the `cdpPos` as default as it may have ill consequences when passed to CDP. Should we use empty string instead? Or maybe we should actually rethrow?
```suggestion
      throw new Error(`Invalid POS format: ${error}`);
```

Reply:
ended up on rethrowing approach - we don't want to invalid data to be passed to an event, but we also don't want to halt the whole app over it - so error is handled in script component.